python/pyspark/sql/dataframe.py

```diff
@@ -859,14 +905,20 @@ def filter(self, condition):
         [Row(age=5, name=u'Bob')]
         >>> df.where("age = 2").collect()
         [Row(age=2, name=u'Alice')]
+
+        >>> df.filter(lambda row: row.age > 3).collect()
+        [Row(age=5, name=u'Bob')]
+        >>> df.map(lambda row: row.age).filter(lambda age: age > 3).collect()
```

Comment:
What's the type of `df.map(lambda row: row.age)`? It's a DataFrame of StructType(BinaryType) or IntegerType?

This looks confusing to me.